src/libYARP_OS/include/yarp/os/Timer.h

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2017 iCub Facility
+ * Authors: Andrea Ruzzenenti
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ */
+
+#include <yarp/os/RateThread.h>
+#include <yarp/os/Mutex.h>
+#include <functional>
+#include <chrono>
+
+namespace yarp {
+    namespace os {
+        class Timer;
+        struct YarpTimerEvent;
+        struct TimerSettings;
+    }
+}
+
+struct yarp::os::YarpTimerEvent
+{   
+    /**
+     * @brief lastExpected when the last callback actually happened
+     */
+    double lastExpected;
+
+    /**
+     * @brief lastReal when the last callback actually happened
+     */
+    double lastReal;
+
+    /**
+     * @brief currentExpected this is when the current callback should have been called
+     */
+    double currentExpected;
+
+    /**
+     * @brief currentReal When the current callback is actually being called
+     */
+    double currentReal;
+
+    /**
+     * @brief lastDuration Contains the duration of the last callback
+     */
+    double lastDuration;
+
+    /**
+     * @brief runCount the count of calls
+     */
+    unsigned int runCount;
+};
+
+struct yarp::os::TimerSettings
+{
+    TimerSettings(std::chrono::milliseconds inRate) : rate(inRate), totalTime(0.0), totalRunCount(0), tollerance(0.001){}
```

Comment:
I don't be the grammar nazi but, it is "tolerance" 😬

Reply:
i make this mistake every time..